.github/workflows/ci-build-binaries.yml

```diff
@@ -130,7 +130,7 @@ jobs:
           cat .goreleaser.yml \
             | go run github.com/t0yv0/goreleaser-filter@v0.3.0 -goos ${{ inputs.os }} -goarch ${{ inputs.arch }} \
             | goreleaser release -f - -p 5 --skip-validate --clean --snapshot
-      - uses: actions/upload-artifact@v2
+      - uses: actions/upload-artifact@v4
```

Comment:
I think this needs `overwrite: true`

Reply:
Pushed here https://github.com/pulumi/pulumi/pull/16485/commits/ab9be11e8fb9abb57d2ccf686129d61012a7951c